browser/net/brave_localhost_permission_network_delegate_helper.cc

```diff
@@ -120,10 +131,15 @@ int OnBeforeURLRequest_LocalhostPermissionWork(
     }
 
     case blink::mojom::PermissionStatus::ASK: {
-      permission_controller->RequestPermissionsFromCurrentDocument(
-          {blink::PermissionType::BRAVE_LOCALHOST_ACCESS},
-          /* rfh */ contents->GetPrimaryMainFrame(), true,
-          base::BindOnce(&OnPermissionRequestStatus, ctx->frame_tree_node_id));
+      // Check if website is allowed to ask for permission.
+      if (g_brave_browser_process->localhost_permission_allowlist_service()
```

Comment:
better to check for nullptr, it can be null is some other tests where the testing browser process is used?